barretenberg/cpp/src/barretenberg/honk/flavor/goblin_translator.hpp

```diff
@@ -0,0 +1,1630 @@
+#pragma once
+#include "../sumcheck/relation_definitions_fwd.hpp"
+#include "barretenberg/ecc/curves/bn254/bn254.hpp"
+#include "barretenberg/honk/pcs/commitment_key.hpp"
+#include "barretenberg/honk/pcs/kzg/kzg.hpp"
+#include "barretenberg/polynomials/univariate.hpp"
+#include "barretenberg/proof_system/arithmetization/arithmetization.hpp"
+#include "barretenberg/proof_system/circuit_builder/goblin_translator_circuit_builder.hpp"
+#include "barretenberg/proof_system/flavor/flavor.hpp"
+#include "barretenberg/proof_system/relations/decomposition_relation.hpp"
+#include "barretenberg/proof_system/relations/extra_relations.hpp"
+#include "barretenberg/proof_system/relations/gen_perm_sort_relation.hpp"
+#include "barretenberg/proof_system/relations/non_native_field_relation.hpp"
+#include "barretenberg/proof_system/relations/permutation_relation.hpp"
+#include <array>
+#include <concepts>
+#include <span>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+namespace proof_system::honk::flavor {
+
+template <size_t mini_circuit_size> class GoblinTranslator_ {
+
+  public:
+    /**
+     * @brief Enum containing IDs of all the polynomials used in Goblin Translator
+     *
+     * @details We use the enum for easier updates of structure sizes and for cases where we need to get a particular
+     * polynomial programmatically
+     */
+    enum ALL_ENTITIES_IDS : size_t {
+        /*The first 4 wires contain the standard values from the EccOpQueue*/
+        OP,
+        X_LO_Y_HI,
+        X_HI_Z_1,
+        Y_LO_Z_2,
+        /*P.xₗₒ split into 2 68 bit limbs*/
```

Comment:
Is the 68 bits a fixed parameter or can it vary? Instead of using a magic number, could you replaced with a name constant?

Reply:
A lot has to change in the code to switch from 68-bit limbs. I'll replace with the name